unit_tests/sources/declarative/parsers/test_model_to_component_factory.py

```diff
@@ -2200,21 +2213,25 @@ def test_add_fields_no_value_type(self):
                         - path: ["field1"]
                           value: "static_value"
         """
-        expected = [
-            AddFields(
-                fields=[
-                    AddedFieldDefinition(
-                        path=["field1"],
-                        value=InterpolatedString(
-                            string="static_value", default="static_value", parameters={}
-                        ),
-                        value_type=None,
-                        parameters={},
-                    )
-                ],
-                parameters={},
-            )
-        ]
+        expected = (
+            [
+                AddFields(
+                    fields=[
+                        AddedFieldDefinition(
+                            path=["field1"],
+                            value=InterpolatedString(
+                                string="static_value", default="static_value", parameters={}
+                            ),
+                            value_type=None,
+                            parameters={},
+                        )
+                    ],
+                    parameters={},
+                )
+            ]
+            * 2
+        )  # Entity will be present twice as we resolve parameters and add the component to both the record
```

Comment:
That seems odd, right? Why would the transformation be there twice? If seems especially dangerous if we eventually have transformations that are not idempotent

Reply:
Is this what is happening?

- Stream parameters have transformations (T1).
- We resolve parameters and pass transformations to stream and record selector(?). Then Stream has T1 and Record selector receives T1.
- Inside of create_record_selector() we check if current model RecordSelector has transformations and we add it, but, we already received the transformation from stream creation logic.
- We end with record selector transformations [T1, T1] 

Im not sure if I'm following correct logic but I agree it is odd. I would say we should check the transformation is not already in the array or something else that keep us away from duplication.

Reply:
Implementation updated with schema transformation